target/linux/realtek/patches-6.6/710-net-phy-sfp-re-probe-modules-on-DEV_UP-event.patch

```diff
@@ -10,7 +10,18 @@ Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
 
```

Comment:
This patch is [described](https://bootlin.com/blog/sfp-modules-on-a-board-running-linux/) as "a hack to work-around the lack of status reporting from the CPSW MAC: we re-probe SFP modules when an interface is brought up."

I suppose things break without this patch? (Although I wouldn't be too surprised if it didn't have an impact either...)

AFAICT this was never submitted upstream, at least not by this person. The description is also rather brief, so I'm finding it hard to match it to patches related to "cpsw" on linux-netdev. I do wonder if the issue they had was ever resolved in a more proper way (in the MAC driver).

Reply:
I'm not sure exactly what breaks.  But I believe it is related to SFP hotplugging on our dsa driver.  Don't know why that needs this hack though.  Most likely a dsa driver bug.